src/components/AccessForm.vue

```diff
@@ -73,24 +78,54 @@ export default defineComponent({
             trigger: "change"
           }
         ]
-      }
+      },
+      loading: false
     };
   },
+  computed: {
+    ...mapState(["ourSecretKey", "decryptedMessage"]),
+    ...mapGetters(["enclavePublicKey"])
+  },
   methods: {
+    ...mapActions(["requestExecutionToken"]),
+    ...mapMutations(["setDecryptedMsg"]),
     onSubmit(formName: string) {
       (this.$refs[formName] as typeof elForm).validate(
         async (valid: boolean) => {
           if (valid) {
-            const enclavePublicKey = this.$store.getters.enclavePublicKey;
+            this.loading = true;
             // Get the unproxied form data before encrypting, for clarity.
             const data = Object.assign({}, this.form);
-            await encryptJson(data, enclavePublicKey);
+            try {
+              await this.requestExecutionToken(data);
+            } catch (e) {
+              console.log(e);
+              // TODO: Error Handling
+            }
```

Comment:
We could use the `notifyErrors` wrapper here?

```suggestion
            await notifyErrors("Execution token request failed", () => {
              return this.requestExecutionToken(data);
            });
```